components/consumer/profile/Thrifts.js

```diff
@@ -0,0 +1,183 @@
+import { useState, useEffect } from "react";
+import HeaderComponent from "@/components/utility/Header";
+import SubscriptionModal from "@/components/scoped/SubscriptionModal";
+import ButtonComponent from "@/components/utility/Button";
+// import { useUser } from "@/context/UserContext";
+import ModalComponent from "@/components/utility/Modal";
+
+import { NotificationManager } from "react-notifications";
+import debounce from "lodash.debounce";
+
+const ThriftList = ({ toggleThrift, consumerData, setConsumerData }) => {
+  const [dataList, setDataList] = useState([]);
+  const [deleteModal, setDeleteModal] = useState(false);
+  const [deleteLoading, setDeleteLoading] = useState(false);
+  const [activeDeleteId, setActiveDeleteId] = useState(null);
+  useEffect(() => {
+    setDataList(consumerData.ThriftList);
+  }, [consumerData]);
+  const onConfirmDeleteListing = async () => {
+    setDeleteLoading(true);
+    const id = activeDeleteId;
+    try {
+      const res = await fetch(`/api/consumer/profile/thriftList/delete`, {
+        method: "DELETE",
+        headers: { "Content-Type": "application/json" },
+        body: JSON.stringify({
+          id: id,
+        }),
+      });
+      const errorData = await res.json();
+      setDeleteLoading(false);
+      if (res.ok) {
+        const filterList = consumerData.ThriftList.filter(
+          (elem) => elem.id !== id
+        );
+        1;
+        setConsumerData({ ...consumerData, ThriftList: filterList });
+        setDeleteModal(false);
+        // fetchListings();
+        setActiveDeleteId(null);
+        NotificationManager.success(errorData.message);
+      } else {
+        // Handle error
+        const errorData = await res.json();
+        NotificationManager.error(errorData);
+      }
+    } catch (error) {
+      console.error("An error occurred while deleting the listing", error);
+    }
+  };
+  const triggerDeleteModal = (index) => {
+    setActiveDeleteId(index);
+    setDeleteModal(true);
+  };
+  const updateThriftListItem = async (fieldValue, id) => {
+    try {
+      console.log("debounce addFieldUpdate", fieldValue, id);
+      const response = await fetch("/api/consumer/profile/thriftList/update", {
+        method: "PUT",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          isChecked: fieldValue,
+          id: id,
+        }),
+      });
+
+      if (response.ok) {
+        const resp = await response.json();
+        const index = consumerData.ThriftList.findIndex(
+          (elem) => elem.id === id
+        );
+        console.log("index", index);
```

Comment:
remove this

Reply:
removed the unnecessary console from this file